### ISSUE TITLE
average latency incorrectly reported

### ISSUE THREAD

**Issue by ceeaspb** [wg/wrk#114]
I noticed while setting up some benchmarking that the average latency is not correct.
Running a simple test with 1 connection and 1 thread we should see that the value for N in Little's  law should approach 1:

```
N=XR
```
where
N is the number of threads applying the load
X is the throughput in requests per second
R is the residence time or latency here

```
./wrk  -c 1 -t 1 -d 1s http://localhost:8080/plaintext
```

as we are applying 1 thread and 1 connection, a calculated N should tend to 1.
I have written a patch ( https://github.com/ceeaspb/wrk/commit/b144c034d103409872c9098babbaef32ab7fe590  ) to output the avarage latency when calculated from the sum of all the latency measurements and count of completed requests over the entire test duration rather than via the statistics structs.

```
# while true; do ./wrk  -c 1 -t 1 -d 1s http://localhost:8080/plaintext 2>/dev/null |awk '/Latency  /{r_bad=$2}/requests in/{x=$1}/Avg latency/{r=$4;print "x= "x" , r_bad= "r_bad" , r= "r" N_bad= "r_bad*x/1000000" , N= "x*r/1000000}';done
x= 17924 , r_bad= 42.85us , r= 46.00 N_bad= 0.768043 , N= 0.824504
x= 18963 , r_bad= 43.22us , r= 42.00 N_bad= 0.819581 , N= 0.796446
x= 19038 , r_bad= 42.30us , r= 42.00 N_bad= 0.805307 , N= 0.799596
x= 19026 , r_bad= 41.61us , r= 42.00 N_bad= 0.791672 , N= 0.799092
x= 18698 , r_bad= 42.83us , r= 43.00 N_bad= 0.800835 , N= 0.804014
x= 19774 , r_bad= 40.71us , r= 41.00 N_bad= 0.805 , N= 0.810734
x= 18575 , r_bad= 42.32us , r= 43.00 N_bad= 0.786094 , N= 0.798725
x= 18690 , r_bad= 42.40us , r= 43.00 N_bad= 0.792456 , N= 0.80367
x= 18721 , r_bad= 43.04us , r= 43.00 N_bad= 0.805752 , N= 0.805003
x= 19032 , r_bad= 42.02us , r= 42.00 N_bad= 0.799725 , N= 0.799344
x= 16695 , r_bad= 42.66us , r= 50.00 N_bad= 0.712209 , N= 0.83475
x= 19165 , r_bad= 42.96us , r= 42.00 N_bad= 0.823328 , N= 0.80493
...
x= 18628 , r_bad= 42.99us , r= 43.00 N_bad= 0.800818 , N= 0.801004
x= 17259 , r_bad= 44.51us , r= 48.00 N_bad= 0.768198 , N= 0.828432
x= 18254 , r_bad= 130.40us , r= 44.00 N_bad= 2.38032 , N= 0.803176
x= 18364 , r_bad= 366.09us , r= 44.00 N_bad= 6.72288 , N= 0.808016
x= 18786 , r_bad= 41.91us , r= 43.00 N_bad= 0.787321 , N= 0.807798
x= 17945 , r_bad= 45.62us , r= 45.00 N_bad= 0.818651 , N= 0.807525
x= 17824 , r_bad= 68.01us , r= 45.00 N_bad= 1.21221 , N= 0.80208
x= 19557 , r_bad= 41.90us , r= 41.00 N_bad= 0.819438 , N= 0.801837
x= 18646 , r_bad= 43.71us , r= 43.00 N_bad= 0.815017 , N= 0.801778
x= 17107 , r_bad= 36.15ms , r= 49.00 N_bad= 0.618418 , N= 0.838243

```

N should be relatively stable, ie. if R increases then with 1 thread X should drop.
The currently reported N_bad however is all over the place.

** Note: I am not suggesting that using the above parameters for wrk is a recommended way of executing real benchmarks! this is just to simplify the situation to show the defect.
** also percentiles should be used over averages, but in this case it it of use to verify the results using Little's law with averages.

The second related thing to note is that at N=0.8 we are "missing" 20% of the offered load from the load injector. either the load is being offered but some latency and or throughput is being omitted, or the missing amount of load is not being offered to the SUT. but that would be another ticket ( #116 ).

running a similar test with ab produces correct results:

```
# while true;do for n in 10 100 1000 10000;do echo $n;ab -c 1 -n $n  http://localhost:8080/plaintext 2>/dev/null |awk '/Requests p/{x=$4}/across all/{r=$4;print x*r/1000" = "x" * "r/1000}';done;done
10
0.999555 = 1118.07 * 0.000894
100
1.00011 = 2681.25 * 0.000373
1000
0.999875 = 3676.01 * 0.000272
10000
1.00116 = 3588.40 * 0.000279
10
0.99974 = 2590.00 * 0.000386
100
0.998641 = 2829.01 * 0.000353
1000
1.00031 = 3664.13 * 0.000273
10000
0.99867 = 3685.13 * 0.000271

```

**Comment by wg** [comment]
@ceeaspb if you have time could you check the 'next' branch and see if the new approach is sufficient to close this issue? Thanks!

**Comment by wg** [comment]
The next branch has been merged and released as 4.0.0.
